base/channels.jl

```diff
@@ -42,7 +42,12 @@ mutable struct Channel{T} <: AbstractChannel
         if sz < 0
             throw(ArgumentError("Channel size must be either 0, a positive integer or Inf"))
         end
-        new(Condition(), Condition(), :open, Nullable{Exception}(), Vector{T}(0), sz, Vector{Task}(0), Vector{Task}(0), 0)
+        ch = new(Condition(), Condition(), :open, Nullable{Exception}(), Vector{T}(0), sz, 0)
```

Comment:
In case of `pmap` only one channel is allocated during its execution. I suspect the performance improvements you are seeing are from `id_in_procs` and not this change. The `data` field in not used in case of buffered channels. If we are going to keep this change, then we should also leave `data` undefined in the buffered case. Post 0.6 branching we should consider having two separate types for buffered and unbuffered channels since there is little overlap in both the fields and implementations.